syft/pkg/cataloger/binary/classifier_cataloger_test.go

```diff
@@ -1667,8 +1667,7 @@ func Test_Cataloger_ResilientToErrors(t *testing.T) {
 	c := NewClassifierCataloger(DefaultClassifierCatalogerConfig())
 
 	resolver := &panicyResolver{}
-	_, _, err := c.Catalog(context.Background(), resolver)
-	assert.Error(t, err)
+	_, _, _ = c.Catalog(context.Background(), resolver)
```

Comment:
I think instead of deleting the error nil check we should check that it's either no nil, nil, or a specific value. It's true that we survived a panic but not clear what the impact to results is under this test.